aocd/runner.py

```diff
@@ -182,24 +194,42 @@ def main():
     sys.exit(rc)
 
 
-def _timeout_wrapper(f, capture=False, timeout=DEFAULT_TIMEOUT, **kwargs):
+def _timeout_wrapper(
+    f: Callable["_PS", _R],
+    capture: bool = False,
+    timeout: float = DEFAULT_TIMEOUT,
+    *args: "_PS.args",
+    **kwargs: "_PS.kwargs",
+) -> pebble.ProcessFuture:
     # aocd.runner executes the user's solve in a subprocess, so that it can be reliably
     # killed if it exceeds a time limit. you can't do that with threads.
     func = pebble.concurrent.process(daemon=False, timeout=timeout)(_process_wrapper)
-    return func(f, capture, **kwargs)
+    return func(f, capture, *args, **kwargs)
 
 
-def _process_wrapper(f, capture=False, **kwargs):
+def _process_wrapper(
+    f: Callable["_PS", _R],
+    capture: bool = False,
+    *args: "_PS.args",
+    **kwargs: "_PS.kwargs",
+)  -> _R:
     # used to suppress any output from the subprocess, if aoc was invoked with --quiet
     with contextlib.ExitStack() as ctx:
         if capture:
             null = ctx.enter_context(open(os.devnull, "w"))
             ctx.enter_context(contextlib.redirect_stderr(null))
             ctx.enter_context(contextlib.redirect_stdout(null))
-        return f(**kwargs)
+        return f(*args, **kwargs)
+    assert False, "unreachable"
 
-
-def run_with_timeout(entry_point, timeout, progress, dt=0.1, capture=False, **kwargs):
+def run_with_timeout(
+    entry_point: EntryPoint,
+    timeout: float,
+    progress: Optional[str],
+    dt: float = 0.1,
+    capture: bool = False,
+    **kwargs: object
```

Comment:
Do not use `object` here, this is definitely a place to use `Any`.

```suggestion
    **kwargs: Any
```

See the [typing best practices documentation on `Any` vs. `object`](https://typing.readthedocs.io/en/latest/source/best_practices.html). Here `kwargs` are keyword arguments passed to `_timeout_wrapper` and you can't express what types are acceptable to the function that is obtained by `entry_point.load()` inside this function.